src/main/java/de/muenchen/isi/domain/mapper/SearchDomainMapper.java

```diff
@@ -1,73 +1,227 @@
 import de.muenchen.isi.infrastructure.entity.BaseEntity;
 import de.muenchen.isi.infrastructure.entity.Baugenehmigungsverfahren;
 import de.muenchen.isi.infrastructure.entity.Bauleitplanverfahren;
 import de.muenchen.isi.infrastructure.entity.Bauvorhaben;
 import de.muenchen.isi.infrastructure.entity.WeiteresVerfahren;
+import de.muenchen.isi.infrastructure.entity.common.Adresse;
+import de.muenchen.isi.infrastructure.entity.common.VerortungMultiPolygon;
 import de.muenchen.isi.infrastructure.entity.infrastruktureinrichtung.Infrastruktureinrichtung;
 import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.ObjectUtils;
+import org.mapstruct.AfterMapping;
 import org.mapstruct.Mapper;
 import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
 import org.mapstruct.Mappings;
 import org.mapstruct.SubclassMapping;
+import org.springframework.beans.factory.annotation.Autowired;
 
+@Slf4j
 @Mapper(
     config = MapstructConfiguration.class,
-    uses = { AbfrageDomainMapper.class, InfrastruktureinrichtungDomainMapper.class, BauvorhabenDomainMapper.class }
+    uses = {
+        AbfrageDomainMapper.class,
+        InfrastruktureinrichtungDomainMapper.class,
+        BauvorhabenDomainMapper.class,
+        KoordinatenDomainMapper.class,
+    }
 )
-public interface SearchDomainMapper {
+public abstract class SearchDomainMapper {
+
+    @Autowired
+    private KoordinatenDomainMapper koordinatenDomainMapper;
+
+    @Autowired
+    private KoordinatenService koordinatenService;
+
     @Mapping(target = "type", ignore = true)
     @SubclassMapping(source = Infrastruktureinrichtung.class, target = InfrastruktureinrichtungSearchResultModel.class)
     @SubclassMapping(source = Bauvorhaben.class, target = BauvorhabenSearchResultModel.class)
     @SubclassMapping(source = Bauleitplanverfahren.class, target = AbfrageSearchResultModel.class)
     @SubclassMapping(source = Baugenehmigungsverfahren.class, target = AbfrageSearchResultModel.class)
     @SubclassMapping(source = WeiteresVerfahren.class, target = AbfrageSearchResultModel.class)
-    SearchResultModel entity2SearchResultModel(final BaseEntity entity);
+    public abstract SearchResultModel entity2SearchResultModel(final BaseEntity entity);
 
     @Mappings(
         {
             @Mapping(target = "type", constant = SearchResultType.Values.BAUVORHABEN),
             @Mapping(source = "verortung.stadtbezirke", target = "stadtbezirke"),
+            @Mapping(target = "coordinate", ignore = true),
         }
     )
-    BauvorhabenSearchResultModel entity2SearchResultModel(final Bauvorhaben entity);
+    public abstract BauvorhabenSearchResultModel entity2SearchResultModel(final Bauvorhaben entity);
+
+    @AfterMapping
+    public void afterEntity2SearchResultModel(
+        final Bauvorhaben entity,
+        @MappingTarget final BauvorhabenSearchResultModel model
+    ) {
+        if (hasAdressCoordinate(entity.getAdresse())) {
```

Comment:
Der Code in den Mapping-Methoden `afterEntity2SearchResultModel` für das Bauvorhaben und der Abfragen ist der selbe.

Die Logik in eine eigene Methode auslagern und wiederverwenden.